cpp/src/reductions/scan/scan_inclusive.cu

```diff
@@ -142,12 +142,13 @@ struct scan_dispatcher {
     return output_column;
   }
 
-  // for string type: only MIN and MAX are supported
-  template <typename T, std::enable_if_t<is_string_supported<T>()>* = nullptr>
-  std::unique_ptr<column> inclusive_scan(column_view const& input_view,
-                                         null_policy,
-                                         rmm::cuda_stream_view stream,
-                                         rmm::mr::device_memory_resource* mr)
+  template <typename T>
+  std::enable_if_t<is_supported<T>() and std::is_same_v<T, cudf::string_view>,
+                   std::unique_ptr<column>>
+  inclusive_scan(column_view const& input_view,
```

Comment:
Maybe combine this and the non-string path by just SFINAE'ing on the is_supported and then do an `if constexpr` on string/not-string internally?  Or potentially just get rid of the SFINAE entirely and just a couple of if constexprs internally.

Reply:
Why is that better? There is not a single line of code in common between these two functions. I like the SFINAE approach here. Is SFINAE no longer preferred?

Reply:
It seems that way.  From what I gather, if constexpr > specialization > SFINAE.  I don't have a strong opinion here though.

Reply:
I reworked this so there is a template-specialization for `cudf::string_view` which gets rid of the SFINAE and does not require `if constexpr`.